src/metatensor/models/utils/omegaconf.py

```diff
@@ -154,3 +154,42 @@ def expand_dataset_config(conf: Union[str, DictConfig]) -> DictConfig:
             )
 
     return conf
+
+
+def check_units(actual_options, desired_options):
+    """Check if the units in the 2 input dataset options are consistent.
```

Comment:
Also check is the target sections are the same right?

Reply:
the function checks 3 things: 
- the length units are the same
- All the target in `desired_options` are in `actual_options`
- the unit of the target are the same in both dataset options